cpp/src/gandiva/regex_functions_holder_test.cc

```diff
@@ -635,4 +635,93 @@ TEST_F(TestExtractHolder, TestErrorWhileBuildingHolder) {
   execution_context_.Reset();
 }
 
+class TestRegexpLikeHolder : public ::testing::Test {
+ protected:
+  ExecutionContext execution_context_;
+};
+
+TEST_F(TestRegexpLikeHolder, TestRegexpLikeUseNonParameter) {
+  EXPECT_OK_AND_ASSIGN(auto regex_like_holder, RegexpLikeHolder::Make("ast", false, ""));
+  auto& regex_like = *regex_like_holder;
+  std::string source_string = "fast";
+
+  auto ret =
+      regex_like(source_string.c_str(), static_cast<int32_t>(source_string.length()));
+  EXPECT_TRUE(ret);
+
+  source_string = "FAST";
+  ret = regex_like(source_string.c_str(), static_cast<int32_t>(source_string.length()));
+  EXPECT_TRUE(!ret);
+}
+
+TEST_F(TestRegexpLikeHolder, TestRegexLikeHolderUseIParameter) {
+  EXPECT_OK_AND_ASSIGN(auto regex_like_holder, RegexpLikeHolder::Make("ast", true, "i"));
```

Comment:
It seems we only have tests covering the `i`/`n`/`m` option, and we could add some more tests to cover:
1) the `c` option seems not covered
2) it will be great if we could add some tests to cover invalid option input as well (non `icmn` option)
3) I am not sure what will/should happen if both `i` and `c` are specified, maybe we could confirm this behavior with a test case

Reply:
For 3: In the oracle documentation, it is described as follows: For example, if you specify 'ic', then Oracle uses case-sensitive matching.